### PR TITLE
incorporate other PRs, finish fixing timer handling

### PR DESCRIPTION
**Pull request by nathwill**
### Description

- stop the timer if timer is disabled
- de-dupe env-file path referencing
- ensure env file exists before service that references it
- restart timer if timer changed

### Issues Resolved

#477

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by nathwill** [comment]
i left out #483 for now; i don't personally have a need for it, and the comment context around the circumstances that call for it seem unlikely to hit production users, so figure if it's needed the request will be resurrected.

File: recipes/systemd_service.rb [hunk 1]
```diff
@@ -69,5 +69,6 @@ class ::Chef::Recipe
       'AccuracySec' => "#{node['chef_client']['splay']}sec",
     }
   )
-  action(timer ? [:create, :enable, :start] : [:disable, :delete])
+  action(timer ? [:create, :enable, :start] : [:stop, :disable, :delete])
+  notifies :restart, to_s, :delayed
```
Comment (iennae) [inline recipes/systemd_service.rb]: this looks like it's missing something with the to_s ? maybe I'm missing something here.

Comment (nathwill) [inline recipes/systemd_service.rb]: `self` is implicit, so the to_s call in this context returns `"systemd_unit[chef client.timer]"`

Comment (iennae) [inline recipes/systemd_service.rb]: ahh interesting. learned something new there.

**Comment by tas50** [comment]
@nathwill any reason not to cut a release at this point?

**Comment by nathwill** [comment]
@tas50 none i know of!

**Pull request merged**